search/downloader_search_callback.hpp

```diff
@@ -25,7 +25,7 @@ class DownloaderSearchCallback
   public:
     virtual ~Delegate() = default;
 
-    virtual void RunUITask(function<void()> fn) = 0;
```

Comment:
Should a UI task be performed on the UI thread? If so would it be better to use platform::SafeCallback insted of std::function here?

Reply:
It's possible, but from my point of view complicates testing of the code, because after switching to SafeCallback Delegate will become useless, and it'll be quite easy in tests to forget to mock gui thread.